_posts/2015-03-stitch-fix-heart-unix.md

```diff
@@ -0,0 +1,670 @@
+
+<a name="systems"></a>
+## Systems
+
+<a name="display-resource-usage-and-availability"></a>
+### Display Resource Usage and Availability
+&#8212; [Eli Bressert](https://twitter.com/astrobiased/) *(OS X / zsh)*
+
+#### Problem
+What is my system's resource usage and availability?
+
+#### Solution
+Use [`htop`](http://hisham.hm/htop/); an interactive process viewer.
+
+#### Install
+`brew install htop`
+
+##### Command
+```
+htop
+```
+
+##### Output
+```
+  1  [|||||||                                           10.5%]     Tasks: 187 total, 0 running
+  2  [|                                                  0.6%]     Load average: 1.72 1.58 1.51
+  3  [||||||||                                          13.2%]     Uptime: 2 days, 03:32:02
+  4  [|                                                  0.7%]
+  5  [||||                                               5.2%]
+  6  [                                                   0.0%]
+  7  [|||||                                              6.5%]
+  8  [                                                   0.0%]
+  Mem[|||||||||||||||||||||||||||||||||||        8369/16384MB]
+  Swp[                                               0/1024MB]
+
+  PID USER     PRI  NI  VIRT   RES   SHR S CPU% MEM%   TIME+  Command
+75871 simeon    31   0 2407M  2144     0 C  0.0  0.0  0:00.00 htop
+
+```
+
+<a name="easily-manage-python-virtual-environments"></a>
+### Easily Manage Python Virtual Environments
+
+&#8212; [Jeff Magnusson](https://www.linkedin.com/in/jmagnuss) *(OS X / bash)*
+
+#### Problem
+Managing multiple Python virtual environments is tedious.
+
+#### Solution
+Install `pyenv` to your `~/.bashrc`.
+
+#### Script
+```bash
+PYTHON_VIRTUALENV_BASEPATH="$HOME/python/virtualenv"
+
+function pyenv {
+    if [ -z "$1" ]; then
+        echo `ls $PYTHON_VIRTUALENV_BASEPATH`
+    elif [ $1 == '--create' ]; then
+        pushd $PYTHON_VIRTUALENV_BASEPATH; virtualenv --no-site-packages $2; popd;
+    else
+        source $PYTHON_VIRTUALENV_BASEPATH/$1/bin/activate;
+    fi
+}
+```
+
+`pyenv` is a light wrapper around Python's `virtualenv` command.  Executed with no arguments (`pyenv`), it returns a list of currently installed virtual environments.  
+
+Executed with a single argument, it attempts to activate the `virtualenv` passed as the argument (`pyenv my_virtual_env`).  
+
+Executed with `--create` flag, it creates the virtual environment passed in the second argument (`pyenv --create my_virtual_env`).
+
+<a name="productivity"></a>
+## Productivity
+
+<a name="alias-everything"></a>
+### Alias Everything
+&#8212; [Eric Gravert](https://twitter.com/egravert) *(OS X, Linux / bash)*
+
+#### Problem
+I hate typing. 
+
+#### Solution
+I alias EVERYTHING.
+
+##### Command
+```
+alias be="bundle exec"
+alias brake="be rake"
+alias bspec="be rspec"
+alias clock='date "+DATE: %Y-%m-%d%nTIME: %r"'
+```
+
+##### Problem
+Typing an alias is still too many keystrokes.
+
+##### Solution
+Create custom key bindings.
```

Comment:
I wrote my suggestions in haste. Maybe expand this to:

`You can create custom key bindings in bash which can be used to execute commands and custom scripts, saving valuable keystrokes. `